jgrapht-core/src/main/java/org/jgrapht/alg/matching/EdmondsMaximumCardinalityMatching.java

```diff
@@ -547,19 +544,87 @@ public boolean isMaximumMatching(Matching<V, E> matching)
                 / 2.0;
     }
 
+    
+    /**
+     * Storage of the forest, even and odd levels.
+     * 
+     * We explicitly maintain a dirty mark in order to be able to cleanup only the 
```

Comment:
Are you certain that just resetting the dirty bits as opposed to resetting the entire even/odd arrays is sufficient? I can't remember the exact details, but could it be the case that not resetting the entire array leads to errors? E.g. if you labeled a vertex in the previous augmenting round which should receive a different label in the new round?

Reply:
We call `levels.reset()` in each round, which means that the even/odd arrays always start as completely initialized.